ietf/doc/utils.py

```diff
@@ -1317,3 +1318,33 @@ def fuzzy_find_documents(name, rev=None):
 
     FoundDocuments = namedtuple('FoundDocuments', 'documents matched_name matched_rev')
     return FoundDocuments(docs, name, rev)
+
+def bibxml_for_draft(doc, rev=None):
+
+    if rev != None and rev != doc.rev:
+        # find the entry in the history
+        for h in doc.history_set.order_by("-time"):
+            if rev == h.rev:
+                doc = h
+                break
+    if rev and rev != doc.rev:
+        raise Http404("Revision not found")
```

Comment:
I spent a few moments pondering raising this from a utility function, and landed on it being the right thing to do still.

Reply:
Your comment was separated from the line by a Codecov comment so I didn't connect it until after I'd already commented more or less the same thing. (I think it's slightly yuck, but we import django everywhere anyway and it takes so much boilerplate to convert it to a 404 in the view that I agree it's preferable to do it this way.)